R/plotBubble.R

```diff
@@ -24,11 +25,9 @@ plotBubble <- function(inSCE, useAssay="logcounts", feature, displayName=NULL, c
 }
 
 .ggBubble <- function(avgExpr, percExpr, colorLow="white", colorHigh="blue", title=""){
-  x <- NULL
-  y <- NULL
-  df <- data.frame(x=avgExpr, y=percExpr)
-  gg <- ggplot2::ggplot(df, ggplot2::aes(x=x.Gene, y=y.cluster)) +
-    ggplot2::geom_point(ggplot2::aes(color=x.clusterAveExpr, size=y.clusterExprPerc)) +
+  df <- data.frame(avgExpr, percExpr)
+  gg <- ggplot2::ggplot(df, ggplot2::aes(x = .data[["Gene"]], y = .data[["cluster"]])) +
```

Comment:
Shouldn't this line just be:

```
gg <- ggplot2::ggplot(df, ggplot2::aes(x = .data[["avgExpr"]], y = .data[["percExpr"]])) +
```

There are no variables called "Gene" or "cluster" within this function any more now that you have changed the input.

Also, make sure to test out the function locally to make sure it works.

Reply:
I believe my line is correct as I referenced the original code for the function, and testing locally works, producing the same output. The dataframe has rows called "Gene" and "Cluster" which I'm referencing here, while "avgExpr" and "percExpr" are each their own dataframes and the df variable has no columns for these names. When I test locally using "avgExpr" and "percExpr" I subsequently get errors since the columns don't exist.

Here is what the dateframe looks like for reference:
> df
  Gene      cluster clusterAveExpr Gene.1    cluster.1 clusterExprPerc
1  B2M      Doublet      94.311111    B2M      Doublet       1.0000000
2  B2M EmptyDroplet       2.517949    B2M EmptyDroplet       0.7282051
3  B2M      Singlet      54.473333    B2M      Singlet       1.0000000

Reply:
Hmm, I'm not sure I fully understand just by looking at the code. Where is "Gene" and "Cluster" defined? Within the scope of .ggBubble, I only see avgExpr and percExpr passed as parameters and then combined into a df. I don't see where Gene and Cluster are within the function (or how the df would have Gene/Cluster). Make sure to clear your environment of any variables with the same name before testing:

```
rm(df, Gene, Cluster)
```

R is a bit weird in that regard. If it doesn't find a variable within the local scope of a function, it will look for it within your broader environment. Maybe another question what are "metrics$avgExpr" and metrics$percExpr? Are they vectors or dfs?